pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -678,6 +699,81 @@ public SuccessResponse reloadAllSegments(
     return new SuccessResponse(JsonUtils.objectToString(perTableMsgData));
   }
 
+  @POST
+  @Path("segments/{tableName}/batchReload")
+  @Authorize(targetType = TargetType.TABLE, paramName = "tableName", action = Actions.Table.RELOAD_SEGMENT)
+  @Authenticate(AccessType.UPDATE)
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Reload with a map from instances to segments for more flexible control", notes = "Reload "
+      + "with a map from instances to segments for more flexible control")
+  public SuccessResponse reloadSegments(
```

Comment:
Consider integrating this into the table reload API (`POST segments/{tableName}/reload`)

Reply:
Should be doable. I didn’t try harder as the new batch reload API submits multiple reload jobs to track reload progress on each instance, as each instance may work in different set of segments.